File: Example/Objective-C/MZFormSheetPresentationController Objective-C Example/PresentedTableViewController.m
//
//  PresentedTableViewController.m
//  MZFormSheetPresentationViewController Objective-C Example
//
//  Created by Michal Zaborowski on 18.06.2015.
//  Copyright (c) 2015 Michal Zaborowski. All rights reserved.
//

#import "PresentedTableViewController.h"
#import "MZFormSheetPresentationViewController.h"

@interface PresentedTableViewController ()

@end

@implementation PresentedTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"Close" style:UIBarButtonItemStylePlain target:self action:@selector(close)];
    

    self.textField.text = self.passingString;
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    if (self.textFieldBecomeFirstResponder) {
        [self.textField becomeFirstResponder];
    }
}

- (void)close {
    [self dismissViewControllerAnimated:YES completion:nil];
}

@end

File: Example/Swift/MZFormSheetPresentationController Swift Example/PresentedTableViewController.swift
//
//  PresentedTableViewController.swift
//  MZFormSheetPresentationController Swift Example
//
//  Created by Michal Zaborowski on 18.06.2015.
//  Copyright (c) 2015 Michal Zaborowski. All rights reserved.
//

import UIKit

class PresentedTableViewController: UITableViewController {

    @IBOutlet weak var textField: UITextField!
    var textFieldBecomeFirstResponder: Bool = false
    var passingString: String?
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.navigationItem.leftBarButtonItem = UIBarButtonItem(title: "Close", style: .plain, target: self, action: #selector(PresentedTableViewController.close))
        
        if let text = self.passingString {
            self.textField.text = text;
        }
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        if textFieldBecomeFirstResponder {
            self.textField.becomeFirstResponder()
        }
    }
    
    @objc func close() -> Void {
        self.dismiss(animated: true, completion: nil)
    }

}
